sw/device/tests/sha256_test.c

```diff
@@ -19,27 +18,22 @@ static const uint32_t kExpectedDigest[8] = {0xdc96c23d, 0xaf36e268, 0xcb68ff71,
                                             0xe92f76e2, 0xb8a8379d, 0x426dc745,
                                             0x19f5cff7, 0x4ec9c6d6};
 
-int main(int argc, char **argv) {
-  uart_init(kUartBaudrate);
-  uart_send_str("Running SHA256 test\r\n");
+bool test_main(void) {
+  LOG_INFO("Running SHA256 test");
 
   uint32_t digest[8];
   hw_SHA256_hash(kData, kDataLen, (uint8_t *)digest);
 
   bool has_error = false;
   for (uint32_t i = 0; i < 8; i++) {
     if (digest[i] != kExpectedDigest[i]) {
+      LOG_ERROR("Digest mismatched at index %d: exp: %x, act: %x", i, digest[i],
+                kExpectedDigest[i]);
       flash_write_scratch_reg(digest[i]);
       has_error = true;
       break;
     }
   }
 
-  if (has_error) {
-    uart_send_str("FAIL!\r\n");
-  } else {
-    uart_send_str("PASS!\r\n");
-  }
-
-  return 0;
+  return !has_error;
```

Comment:
Consider phrasing this as `return true;` and replacing `has_error = true; break;` with `return false;`.

Reply:
Done.

Reply:
@cdgori, this style is probably okay for DV / testing. Agree?

Reply:
@cdgori Interesting, I didn't know about that. I don't think that's a concern for tests, and if it's a concern for our libraries we should probably discuss it further in an issue. (We already make extensive use of early return in DIF libraries, for example.)

Reply:
I filed #2179 to track this.

Reply:
@sriyerg agree for this usage (DV/testing).